storage/range_raftstorage.go

```diff
@@ -348,8 +348,12 @@ func (r *Range) updateRangeInfo() error {
 		return util.Errorf("failed to lookup zone config for Range %s: %s", r, err)
 	}
 	r.SetMaxBytes(zone.RangeMaxBytes)
-	// TODO(kkaneda): Update configHashes and other fields as well (#1362)?
 
+	// Gossip configs and update configHashes if a lease is held. This is normally unnecessary
```

Comment:
Agreed. It's not probably worth copying configHashes as the reason you described.

I looked at how other range' in-memory states are managed:

```
// Derived from data stored in disk.
// Set when a range is split or a snapshot is applied.
desc
lease
lastIndex
appliedIndex

// Derived from gossip information.
// Need to be set when a range is split or a snapshot is applied
// since this will not be updated until the value changes.
maxBytes int64

// Derived from config data stored in disk.
// No need to be updated from ApplySnapshot or SplitRange
// since it will be anyway set when maybeGossipConfigs is called.
configHashes

// Updated whenever a Raft command is applied. Each replica keeps
// the exactly same copy of the data.
//
// splitTrigger copies this to a new range, but ApplySnapshot does
// not.
respCache

// No need to sync. Only used by a leader replica. The low
// watermark is updated when a new lease is acquired so
// that old data do not need to be referenced.
// splitTrigger copies this to a new range, but ApplySnapshot does not.
tsCache

// No need to sync. They are used to keep track of commands submitted
// to a specific replica.
cmdQ
pendingCmds

// No need to be copied since it's anyway stats info.
// splitTrigger recomputes this for a new range. ApplySnapshot also
// copies it.
stats    *rangeStats
```

Do you think `respCache` and `tsCache` need to be updated from `ApplySnapshot`? I'm wondering `respCache` needs to be, but `tsCache` doesn't.